services/gateway/shared/registerHttpApi.js

```diff
@@ -64,137 +79,148 @@ const configureApi = (apiNames, apiPrefix, registeredModuleNames) => {
+			res.end(data.data);
+			return res;
+		}
+
+		if (data.data && data.status) {
+			if (data.status === 'SERVICE_UNAVAILABLE') ctx.meta.$responseHeaders = { 'Retry-After': 30 };
+			if (data.status === 'INVALID_PARAMS') data.status = 'BAD_REQUEST';
+
+			ctx.meta.$statusCode = StatusCodes[data.status] || data.status;
+
+			// 'ACCEPTED' is a successful HTTP code
+			if (data.status !== 'ACCEPTED') {
+				let message = `The request ended up with ${data.status} error`;
+
+				if (typeof data.data === 'object' && typeof data.data.error === 'string') {
+					message = data.data.error;
+				}
+
+				return {
 					error: true,
 					message,
-				}));
-			};
-
-			const routeAlias = `${req.method.toUpperCase()} ${req.$alias.path}`;
-			const paramReport = validate(req.$params, methodPaths[routeAlias]);
-
-			if (paramReport.missing.length > 0) {
-				sendResponse(INVALID_REQUEST[0], `Missing parameter(s): ${paramReport.missing.join('; ')}`);
-				throw new ValidationException('Request param validation error.');
-			}
-
-			const unknownList = Object.keys(paramReport.unknown);
-			if (unknownList.length > 0) {
-				sendResponse(INVALID_REQUEST[0], `Unknown input parameter(s): ${unknownList.join('; ')}`);
-				throw new ValidationException('Request param validation error.');
+				};
 			}
+		}
+		const apiPath = `${req.method.toUpperCase()} ${req.$alias.path}`;
+		return transformResponse(methodPaths[apiPath], data);
+	},
+});
 
-			if (paramReport.required.length) {
-				sendResponse(INVALID_REQUEST[0], `Require one of the following parameter combination(s): ${paramReport.required.join('; ')}`);
-				throw new ValidationException('Request param validation error.');
-			}
-
-			const invalidList = paramReport.invalid;
-			if (invalidList.length > 0) {
-				sendResponse(INVALID_REQUEST[0], `Invalid input: ${invalidList.map(o => o.message).join('; ')}`);
-				throw new ValidationException('Request param validation error.');
-			}
-
-			const params = transformRequest(methodPaths[routeAlias], dropEmptyProps(paramReport.valid));
-			req.$params = params;
-		},
-
-		async onAfterCall(ctx, route, req, res, data) {
-			if (gatewayConfig.api.enableHttpCacheControl) {
-				// Set 'Cache-Control' to assist response caching
-				ctx.meta.$responseHeaders = { 'Cache-Control': gatewayConfig.api.httpCacheControlDirectives };
-			}
-
-			// Set response headers and return CSV data if filename available
-			if (data.data && data.meta && data.meta.filename && data.meta.filename.endsWith('.csv')) {
-				res.setHeader('Content-Disposition', `attachment; filename="${data.meta.filename}"`);
-				res.setHeader('Content-Type', 'text/csv');
-				res.end(data.data);
-				return res;
+const registerApi = (apiNames, config, registeredModuleNames) => {
+	const allAPIs = getAllAPIs(apiNames, registeredModuleNames);
+
+	const apisToRegister = [];
+
+	const DEFAULT_ETAG_VALUE = 'strong';
+	const ALLOWED_ETAG_VALUES = ['strong', 'weak', true, false, 'custom'];
```

Comment:
Please move these to the global level. Also, define DEFAULT_ALIAS and DEFAULT_METHOD_PATH to be used in `buildAPIAliases`.